package fantasybit;

message FantasyNameHash {
	optional string name = 10;
	optional uint64 hash = 30;
}

message Distribution {
	optional string gameid = 2;
	optional string teamid = 4;
	optional int32 season = 5;
	optional int32 week = 6;
        optional uint64 fantasy_nameid = 1;
        optional string playerid = 3;
        optional int32 proj = 7;
        optional float award = 8;
        optional float result = 9;
}

message Profits {
        optional string gameid = 2;
        optional string teamid = 4;
        optional int32 season = 5;
        optional int32 week = 6;
        optional uint64 fantasy_nameid = 1;
        optional string playerid = 3;
        optional int32 qty = 7;
        optional float price = 8;
        optional float result = 9;
        optional float pnl = 10;
}

message Fills {
    optional bool passivebuy = 1;
    optional string buyer = 10;
    optional string seller = 20;
    optional string teamid = 40;
    optional string playerid = 45;
    optional int32 season = 50;
    optional int32 week = 60;
    optional int32 qty = 70;
    optional float price = 80;

}

message ProjKstats {
    optional float pa = 10;
    optional float fg30 = 20;
    optional float fg40 = 30;
    optional float fg50 = 40;
    optional float fg60 = 50;
}

message ProjDstats {
    optional float twopt = 100;
    optional float onept = 110;
    optional float deftd = 120;
    optional float sfty = 130;
    optional float turnovers = 140;
    optional float sacks = 150;
    optional float ptsa = 160;
}

message ProjOstats {
    optional float passyds = 10;
    optional float passtd = 20;
    optional float rushyds = 30;
    optional float rushtd = 40;
    optional float recyds = 50;
    optional float rectd = 60;
    optional float rec = 70;
    optional float pint = 80;
    optional float fumble = 90;
    optional float twopt = 100;
    optional float onept = 110;
}

message ProjStats {
    optional ProjOstats ostats = 10;
    optional ProjDstats dstats = 20;
    optional ProjKstats kstats = 30;
}


